libzfs.pyx

```diff
@@ -2249,12 +2249,16 @@ cdef class ZFSObject(object):
 
 
 cdef class ZFSDataset(ZFSObject):
-    def __getstate__(self, recursive=True):
+    def __getstate__(self):
         ret = super(ZFSDataset, self).__getstate__()
-        ret['mountpoint'] = self.mountpoint
 
-        if recursive:
-            ret['children'] = [i.__getstate__() for i in self.children]
+        # TODO: Talk to William, probably returning snapshots here wouldn't be desired because of the performance drop
```

Comment:
Exactly, we should keep recursive option there, and add a new one for snapshots (default to False)